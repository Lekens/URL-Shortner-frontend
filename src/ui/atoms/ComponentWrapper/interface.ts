import { ReactNode } from 'react';

export interface IComponentWrapper {
  children: ReactNode;
  extraClassName?: string;
}
