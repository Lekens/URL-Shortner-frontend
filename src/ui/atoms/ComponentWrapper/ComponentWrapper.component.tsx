import { useMemo } from 'react';

import cn from 'classnames';

import { IComponentWrapper } from 'ui/atoms/ComponentWrapper/interface';

import styles from './styles.module.scss';

function ComponentWrapper({ children, extraClassName }: IComponentWrapper) {
  const wrapperStyle = useMemo(
    () => cn(styles.children, extraClassName),
    [extraClassName],
  );

  return <div className={wrapperStyle}>{children}</div>;
}
export default ComponentWrapper;
