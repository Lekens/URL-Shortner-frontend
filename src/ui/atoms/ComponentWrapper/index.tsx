export { default as ComponentWrapper } from './ComponentWrapper.component';
