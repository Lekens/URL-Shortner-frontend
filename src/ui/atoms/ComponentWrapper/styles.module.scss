.children {
  padding: 0 10px;
}
