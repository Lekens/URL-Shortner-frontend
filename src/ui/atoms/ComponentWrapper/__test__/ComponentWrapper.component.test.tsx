import React from 'react';

import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { shallow } from 'enzyme';

import ComponentWrapper from 'ui/atoms/ComponentWrapper/ComponentWrapper.component';

describe('Should test ComponentWrapper', () => {
  it('Should match snapshot', () => {
    const state = {
      getState() {},
      subscribe: () => {},
      dispatch: () => {},
    };
    const { container } = render(
      <Provider store={state as any}>
        <ComponentWrapper>
          <div>Hello World</div>
        </ComponentWrapper>
      </Provider>,
    );

    expect(container).toMatchSnapshot();
  });
  it('Should find Hello text on Dom', () => {
    const state = {
      getState() {},
      subscribe: () => {},
      dispatch: () => {},
    };
    const container = shallow(
      <Provider store={state as any}>
        <ComponentWrapper>
          <div>Hello World, This is a Child Dom</div>
        </ComponentWrapper>
      </Provider>,
    );

    expect(
      container.contains(<div>Hello World, This is a Child Dom</div>),
    ).toBe(true);
  });
});
