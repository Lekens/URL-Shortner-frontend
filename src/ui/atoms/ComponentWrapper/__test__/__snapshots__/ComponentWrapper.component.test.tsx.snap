// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should test ComponentWrapper Should match snapshot 1`] = `
<div>
  <div
    class="children"
  >
    <div>
      Hello World
    </div>
  </div>
</div>
`;
